src/shared/components/lollipopMutationPlot/LollipopPlotNoTooltip.tsx

```diff
@@ -0,0 +1,335 @@
+import * as React from 'react';
+import _ from "lodash";
+import Lollipop from "./Lollipop";
+import Domain from "./Domain";
+import SVGAxis from "../SVGAxis";
+import {Tick} from "../SVGAxis";
+import {observer} from "mobx-react";
+import {observable, asMap, ObservableMap, computed} from "mobx";
+import $ from "jquery";
+import {LollipopPlotProps} from "./LollipopPlot";
+
+export type LollipopSpec = {
+    codon:number;
+    count:number;
+    label?:string;
+    color?:string;
+    tooltip?:JSX.Element;
+    onMouseOver?:()=>void;
+    onMouseOut?:()=>void;
+};
+
+export type DomainSpec = {
+    startCodon:number;
+    endCodon:number;
+    color:string;
+    label?:string;
+    labelColor?:string;
+    tooltip?:JSX.Element;
+};
+
+export type LollipopPlotNoTooltipProps = LollipopPlotProps & {
+    setHitZone?:(
+        hitRect:{x:number, y:number, width:number, height:number},
+        tooltipContent?:JSX.Element,
+        onMouseEnter?:()=>void,
+        onClick?:()=>void,
+        onMouseLeave?:()=>void
+    )=>void;
+    onMouseLeave?:()=>void;
+    onBackgroundMouseMove?:()=>void;
+};
+
+const DELETE_FOR_DOWNLOAD_CLASS = "delete-for-download";
+
+@observer
+export default class LollipopPlotNoTooltip extends React.Component<LollipopPlotNoTooltipProps, {}> {
+
+    @observable private selectedPositions:ObservableMap<boolean> = observable.map<boolean>();
+
+    private svg:SVGElement;
+    private shiftPressed:number = 0;
+    private handlers:any;
+
+    private lollipopZeroHeight = 10;
+    private xAxisCandidateTickIntervals = [50, 100, 200, 250, 500, 1000, 2500, 5000, 10000, 25000];
+    private yAxisCandidateTickIntervals = [1,2,5,10,20,50,100,200,500];
+    private xAxisHeight = 30;
+    private yAxisWidth = 50;
+    private geneHeight = 14;
+    private domainHeight = 24;
+
+    constructor() {
+        super();
+        this.handlers = {
+            ref: (svg:SVGElement)=>{this.svg = svg;},
+            onBackgroundClick:()=>{
+                this.selectedPositions.clear();
+                this.props.onSelectionChange && this.props.onSelectionChange(this.selectedPositions.toJS());
+            },
+            onBackgroundMouseMove:()=>{
+                this.props.onBackgroundMouseMove && this.props.onBackgroundMouseMove();
+            },
+            onLollipopClick:(codon:number)=>{
+                const prevSelection = this.selectedPositions.get(codon+"");
+                if (!this.shiftPressed) {
+                    this.selectedPositions.clear();
+                }
+                this.selectedPositions.set(codon+"", !prevSelection);
+                this.props.onSelectionChange && this.props.onSelectionChange(this.selectedPositions.toJS());
+            }
+        };
+    }
+
+
+    componentDidMount() {
+        $(document).keydown(e=>{
```

Comment:
you should kill these listeners on unmount of component. otherwise, i believe there will be a memory leak.